src/test/stemcontrolobjecttest.cpp

```diff
@@ -0,0 +1,254 @@
+    loadTrack(m_pMixerDeck1, pTrack);
+
+    EXPECT_EQ(m_pStemCount->get(), 4.0);
+}
+
+TEST_F(StemControlTest, StemColor) {
+    EXPECT_EQ(m_pStem1Color->get(), 0xfd << 16 | 0x4a << 8 | 0x4a);
+    EXPECT_EQ(m_pStem2Color->get(), 0xff << 16 | 0xff << 8 | 0x00);
+    EXPECT_EQ(m_pStem3Color->get(), 0x00 << 16 | 0xe8 << 8 | 0xe8);
+    EXPECT_EQ(m_pStem4Color->get(), 0xad << 16 | 0x65 << 8 | 0xff);
+
+    QString kTrackLocationTest = getTestDir().filePath(QStringLiteral("sine-30.wav"));
+    TrackPointer pTrack(Track::newTemporary(kTrackLocationTest));
+    loadTrack(m_pMixerDeck1, pTrack);
+
+    EXPECT_EQ(m_pStem1Color->get(), -1.0);
+    EXPECT_EQ(m_pStem2Color->get(), -1.0);
+    EXPECT_EQ(m_pStem3Color->get(), -1.0);
+    EXPECT_EQ(m_pStem4Color->get(), -1.0);
+
+    kTrackLocationTest = getTestDir().filePath("stems/test.stem.mp4");
+    pTrack = Track::newTemporary(kTrackLocationTest);
+    loadTrack(m_pMixerDeck1, pTrack);
+
+    EXPECT_EQ(m_pStem1Color->get(), 0xfd << 16 | 0x4a << 8 | 0x4a);
+    EXPECT_EQ(m_pStem2Color->get(), 0xff << 16 | 0xff << 8 | 0x00);
+    EXPECT_EQ(m_pStem3Color->get(), 0x00 << 16 | 0xe8 << 8 | 0xe8);
+    EXPECT_EQ(m_pStem4Color->get(), 0xad << 16 | 0x65 << 8 | 0xff);
+}
+
+TEST_F(StemControlTest, Volume) {
+    m_pChannel1->getEngineBuffer()->queueNewPlaypos(
+            mixxx::audio::FramePos{0}, EngineBuffer::SEEK_STANDARD);
+    m_pPlay->set(1.0);
+    m_pStem1Volume->set(0.0);
+    m_pStem2Volume->set(0.0);
+    m_pStem3Volume->set(0.0);
+    m_pStem4Volume->set(0.0);
+
+    m_pEngineMixer->process(kMaxEngineChannels * kMaxEngineFrames);
+    assertBufferMatchesReference(m_pEngineMixer->getMainBuffer(),
+            kProcessBufferSize,
+            "StemVolumeControlSilence");
+
+    m_pChannel1->getEngineBuffer()->queueNewPlaypos(
+            mixxx::audio::FramePos{0}, EngineBuffer::SEEK_STANDARD);
+    m_pStem1Volume->set(1.0);
+
+    m_pEngineMixer->process(kMaxEngineChannels * kMaxEngineFrames);
+    assertBufferMatchesReference(m_pEngineMixer->getMainBuffer(),
+            kProcessBufferSize,
+            "StemVolumeControlDrumOnly");
+
+    m_pChannel1->getEngineBuffer()->queueNewPlaypos(
+            mixxx::audio::FramePos{0}, EngineBuffer::SEEK_STANDARD);
+    m_pStem2Volume->set(0.8);
+
+    m_pEngineMixer->process(kMaxEngineChannels * kMaxEngineFrames);
+    assertBufferMatchesReference(m_pEngineMixer->getMainBuffer(),
+            kProcessBufferSize,
+            "StemVolumeControlDrumAndBass");
+
+    m_pChannel1->getEngineBuffer()->queueNewPlaypos(
+            mixxx::audio::FramePos{0}, EngineBuffer::SEEK_STANDARD);
+    m_pStem1Volume->set(0.5);
+    m_pStem3Volume->set(0.2);
+    m_pStem4Volume->set(0.4);
+
+    m_pEngineMixer->process(kMaxEngineChannels * kMaxEngineFrames);
+    assertBufferMatchesReference(m_pEngineMixer->getMainBuffer(),
+            kProcessBufferSize,
+            "StemVolumeControlFull");
+}
+
+TEST_F(StemControlTest, VolumeResetOnLoad) {
```

Comment:
For some reason this test failed here
https://github.com/mixxxdj/mixxx/actions/runs/10981245295/job/30488031679?pr=13077#step:11:1823
(in case the link doesn't work, the PR is #13077)

any idea why @acolombier ?

Reply:
No idea. It looks like it is failing with an unrelated assertion, not quite sure why this would happen

Reply:
oh, I see it's loading one of the dummy tracks. sorry for the noise then